src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxconverter.py

```diff
@@ -0,0 +1,108 @@
+        return source_schema
+
+    @classmethod
+    def get_streams(cls, folder_path: FolderPathType) -> List[str]:
+        from spikeinterface.extractors import SpikeGLXRecordingExtractor
+
+        return SpikeGLXRecordingExtractor.get_streams(folder_path=folder_path)[0]
+
+    def __init__(
+        self,
+        folder_path: FolderPathType,
+        streams: Optional[List[str]] = None,
+        verbose: bool = False,
+    ):
+        """
+        Read all data from multiple streams stored in the SpikeGLX format.
+
+        This can include...
+            (a) single-probe but multi-band such as AP+LF streams
+            (b) multi-probe with multi-band
+            (c) with or without the associated NIDQ channels
+
+        Parameters
+        ----------
+        folder_path : DirectoryPath
+            Path to folder containing the NIDQ stream and subfolders containing each IMEC stream.
+        streams : list of strings, optional
+            A specific list of streams you wish to load.
+            To see which streams are available, run `SpikeGLXConverter.get_streams(folder_path="path/to/spikeglx")`.
+            By default, all available streams are loaded.
+        verbose : bool, default: False
+            Whether to output verbose text.
+        """
+        folder_path = Path(folder_path)
+
+        available_streams = self.get_streams(folder_path=folder_path)
+        streams = streams or available_streams
+
+        data_interfaces = dict()
+        for stream in streams:
+            if "ap" in stream:
+                probe_name = stream[:5]
+                file_path = (
+                    folder_path / f"{folder_path.stem}_{probe_name}" / f"{folder_path.stem}_t0.{probe_name}.ap.bin"
+                )
+                interface = SpikeGLXRecordingInterface(file_path=file_path)
+            if "lf" in stream:
+                probe_name = stream[:5]
+                file_path = (
+                    folder_path / f"{folder_path.stem}_{probe_name}" / f"{folder_path.stem}_t0.{probe_name}.lf.bin"
+                )
+                interface = SpikeGLXLFPInterface(file_path=file_path)
+            if "nidq" in stream:
+                file_path = folder_path / f"{folder_path.stem}_t0.nidq.bin"
+                interface = SpikeGLXNIDQInterface(file_path=file_path)
+                num_channels = interface.recording_extractor.get_num_channels()
+                # To avoid warning/error turing write
+                # TODO: When PyNWB supports other more proper AUX electrode types, remove
+                interface.recording_extractor.set_property(key="shank_electrode_number", values=[np.nan] * num_channels)
+                interface.recording_extractor.set_property(key="contact_shapes", values=[np.nan] * num_channels)
+            data_interfaces.update({stream: interface})
+
+        super().__init__(data_interfaces=data_interfaces, verbose=verbose)
+
+    def get_conversion_options(self) -> dict:
+        return dict()
```

Comment:
```suggestion
```

Reply:
do we need this?

Reply:
Yes, yes we do - as I discovered while debugging

TBH I don't think the method itself is needed anywhere and should probably be removed entirely, but I'm playing by the rules in this PR

Reply:
ok, in that case can you indicate this with a comment in the code?

Reply:
Done